### PR TITLE
[Messenger] Uses an `AmqpStamp` to provide flags and attributes

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28885
| License       | MIT
| Doc PR        | ø

Using the `AmqpStamp` you can configure the flags and any attribute (such as `delivery_mode`).

### CODE REVIEW

File: src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php [hunk 1]
```diff
@@ -206,19 +207,35 @@ public function countMessagesInQueues(): int
...
-            AMQP_NOPARAM,
             [
                 'headers' => $headers,
-            ]
+            ],
+            $amqpStamp
+        );
+    }
+
+    private function publishOnExchange(
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php]: one line please, always for signatures (and lines throwing exceptions btw)

File: src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php [hunk 2]
```diff
@@ -432,4 +449,13 @@ public function purgeQueues()
             $this->queue($queueName)->purge();
         }
     }
+
+    private function routingKeyFromStamp(AmqpStamp $amqpStamp = null): ?string
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php]: I'd suggest removing this method and using the ternary op instead

**Review by nicolas-grekas — APPROVED** [review]
LGTM with minor comments.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @sroze.

**Pull request merged**